### DISCUSSION TITLE
week14 AWS EC2 IP 瀏覽器連線失敗

### DISCUSSION THREAD

**Discussion by weih210287** [Lidemy/forum#202] [程式導師實驗計畫] [answered]
如標題，目前正在 14 週的部署地獄。

目前已經成功開啟了一台 EC2，但當我試著從公有的 IP 查看他預設的 HTML 時卻會出現連線錯誤：

![](https://i.imgur.com/ExmNBBA.png)

![](https://i.imgur.com/QbJrNm7.png)

以下是我目前的執行個體設定：
1. Unbuntu 18.04(或 20.04) LTS (HVM), SSD Volume Type
2. t2.micro
3. 其他都照預設的，沒有更動

- 安全群組
![image](https://user-images.githubusercontent.com/54532629/132190347-f3253ae1-a4d6-4693-8b0d-3213e3133805.png)

自己在電腦上用 PuTTY 有成功進到虛擬機裡，各種操作目前都沒問題，也可以看到預設的 index.html，就是瀏覽器上沒辦法連上網頁。

參考了網上的其他類似的問題如連不上 EC2 網頁等等 [Can't access site on EC2 instance via public ip](https://stackoverflow.com/questions/10253484/cant-access-site-on-ec2-instance-via-public-ip) 得到的答案幾乎都是安全群組沒設好，所以我就試著把所有的端口都設成 `All` 但還是沒辦法：

![](https://i.imgur.com/BuSzYrv.png)

自己目前理解的是 Inbound rules 是指要開放進到執行個體的流量，Outbound rules 是執行個體對外的流量，所以如果要想看得到執行個體裡的 html 檔的話好像是要改 Outbound (嗎？

目前兩邊都改過了還是失敗，不確定是自己哪邊觀念沒有搞清楚或設定出問題，想要請問有沒有同學之前也遇過類似的問題呢？

- 附註
在發問的時候找到解法了，就是把網址從 https 改成 http 後就可以成功連上網站，但我還是不太懂為什麼 https 會失敗，明明已經開放所有端口了為什麼只能用 http 連呢？所以還是想請教大家，謝謝 TvT

**Comment by aszx87410** [comment, marked as answer]
因為你的網站沒有提供 https 服務，所以連不上
https 是要主動設置才有的，不是預設的

**Reply by weih210287** [reply]
原來是這樣，沒有想到是因為根本沒有的關係😅
感謝老師！！
